source/extensions/filters/http/lua/lua_filter.h

```diff
@@ -547,7 +549,14 @@ class Filter : public Http::StreamFilter, private Filters::Common::Lua::LuaLogga
     }
     const Buffer::Instance* bufferedBody() override { return callbacks_->decodingBuffer(); }
     void continueIteration() override { return callbacks_->continueDecoding(); }
-    void onHeadersModified() override { callbacks_->downstreamCallbacks()->clearRouteCache(); }
+    void onHeadersModified() override {
+      // Do not clear route cache if clear_route_cache is false or if no downstream callbacks are
+      // available.
+      if (!parent_.config_->clearRouteCache() || !callbacks_->downstreamCallbacks()) {
```

Comment:
Is there a case where there are no downstream callbacks? We didn't need a check for that previously.

Reply:
See https://github.com/envoyproxy/envoy/pull/38465. The lua filter will be dual filter in the future. So, I did Defensive Programming. 🤣 But I am fine to update it if you think we'd better to leave it as that is.

Reply:
Got it; I'm fine leaving it defensive.